contracts/QuadReader.sol

```
@@ -39,519 +33,300 @@ import "./storage/QuadGovernanceStore.sol";
+        return latest;
+    }
 
-            erc20.safeTransferFrom(msg.sender, address(passport), amountToken);
+    /// @dev Withdraw to  an issuer's treasury or the Quadrata treasury
+    /// @notice Restricted behind a TimelockController
+    /// @param _to address of either an issuer's treasury or the Quadrata treasury
+    /// @param _amount amount to withdraw
+    function withdraw(address payable _to, uint256 _amount) external override {
+        require(
+            IAccessControlUpgradeable(address(governance)).hasRole(GOVERNANCE_ROLE, msg.sender),
+            "INVALID_ADMIN"
+        );
+        require(passport.passportPaused() == false, "Pausable: paused");
+        bool isValid = false;
+        address issuerOrProtocol;
 
-            uint256 amountIssuer = _issuers.length == 0 ? 0 : amountToken * governance.revSplitIssuer() / 10 ** 2;
-            uint256 amountProtocol = amountToken - amountIssuer;
-            for(uint256 i = 0; i < _issuers.length; i++) {
-                passport.increaseAccountBalance(_tokenPayment,governance.issuersTreasury(_issuers[i]), amountIssuer / _issuers.length);
+        if (_to == governance.treasury()) {
+            isValid = true;
+            issuerOrProtocol = address(this);
+        }
+
+        address[] memory issuers = governance.getIssuers();
+        for (uint256 i = 0; i < issuers.length; i++) {
+            if (_to == governance.issuersTreasury(issuers[i])) {
+                isValid = true;
+                issuerOrProtocol = issuers[i];
+                break;
             }
-            passport.increaseAccountBalance(_tokenPayment, governance.treasury(), amountProtocol);
         }
-    }
 
+        require(_to != address(0), "WITHDRAW_ADDRESS_ZERO");
+        require(isValid, "WITHDRAWAL_ADDRESS_INVALID");
+        require(_amount <= address(this).balance, "INSUFFICIENT_BALANCE");
+        (bool sent,) = _to.call{value: _amount}("");
+        require(sent, "FAILED_TO_TRANSFER_NATIVE_ETH");
```

How is this line even testable?

---

not sure. I didn't change the behavior here. What do you suggest?

---

If we did want to test this line, we could withdraw to a contract that doesn't have a fallback. That should raise the error.